src/biogeophys/CanopyHydrologyMod.F90

```diff
@@ -308,12 +310,14 @@ subroutine CanopyHydrology(bounds, &
 
 
              if (col%itype(c) /= icol_sunwall .and. col%itype(c) /= icol_shadewall) then
+                if (frac_veg_nosno(p) == 1 .and. (forc_rain(c) + forc_snow(c) + qflx_irrig_sprinkler(p)) > 0._r8) then
 
-                if (frac_veg_nosno(p) == 1 .and. (forc_rain(c) + forc_snow(c)) > 0._r8) then
-
+                   ! total liquid water inputs above canopy
+                   qflx_liq_above_canopy(p) = forc_rain(c)+ qflx_irrig_sprinkler(p)
+                   
                    ! determine fraction of input precipitation that is snow and rain
-                   fracsnow(p) = forc_snow(c)/(forc_snow(c) + forc_rain(c))
-                   fracrain(p) = forc_rain(c)/(forc_snow(c) + forc_rain(c))
+                   fracsnow(p) = forc_snow(c)/(forc_snow(c) + qflx_liq_above_canopy(p))
+                   fracrain(p) = forc_rain(c)/(forc_snow(c) + qflx_liq_above_canopy(p))
```

Comment:
Is it correct to have forc_rain in the numerator here, or should the numerator be qflx_liq_above_canopy? As written, fracsnow + fracrain may be less than 1. I'm not sure what's right here, but just wanted to double-check.